layersvt/device_simulation.cpp

```diff
@@ -1331,6 +1501,35 @@ VKAPI_ATTR void VKAPI_CALL GetPhysicalDeviceProperties(VkPhysicalDevice physical
     }
 }
 
+// Utility function for iterating through the pNext chain of certain Vulkan structs.
+void FillPNextChain(PhysicalDeviceData *physicalDeviceData, void *place) {
```

Comment:
The only thing that's a little odd is that this will fill in a portability struct passed to the wrong API call. I couldn't find any spec language explicitly stating that it's illegal to do so, so I think this implementation is technically valid. The reason it still concerns me a little is because I wouldn't want applications to start depending on what is technically undefined behavior.

Reply:
This change might be a bit more difficult, so I'll put it off for now, but will fix it in the future.